src/app/core/products/product.service.ts

```diff
@@ -112,7 +112,9 @@ export class ProductService {
     getFeaturedProducts(): Observable<Product[]> {
         return this.getProducts()
             .pipe(
-                map(arr => arr.filter(p => p.collection === "featured")),
+                map(arr =>
+                    arr.filter(p => p.collection === "featured")
+                        .slice(0, 2)),
                 take(2)
```

Comment:
do you need both slice(0,2) and take(2)?

Reply:
So my thinking here was that we're returning an `Observable<Product[]>` which means we need to map the array down to two values using `slice`, but once the back-end is connected, we'll be returning an `Observable[Product]` at which point we need the `take` to return just two values. So yeah, the `take` is pointless at the moment because we only ever produce a single array, but later on, when the whole `map` bit gets removed, the `take` will come in to play and I figured it'd be prudent to leave it there for now so I don't forget about it further down the line. Maybe that's a bit cluttered, though?